### DISCUSSION TITLE
Some CMake projects installing to 'lib64' instead of 'lib' after switching to OpenSUSE Tumbleweed

### DISCUSSION THREAD

**Discussion by salotz** [spack/spack#33357] [Q&A]
Hello,

I've been maintaining some Spack [recipes](https://github.com/salotz/snailpacks) and I recently switched from Ubuntu 20.04 to OpenSUSE Tumbleweed.

```console
$ spack debug report
* **Spack:** 0.18.1 (13e6f87ef6527954b152eaea303841978e83b992)
* **Python:** 3.10.7
* **Platform:** linux-opensusetumbleweed2.36-zen3
* **Concretizer:** clingo
$  gcc --version
gcc (SUSE Linux) 12.2.1 20220830 [revision e927d1cf141f221c5a32574bde0913307e140984]
Copyright (C) 2022 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

Some of the recipes ([this](https://github.com/salotz/snailpacks/tree/master/packages/raylib) & [this](https://github.com/salotz/snailpacks/tree/master/packages/spirv-tools)) started failing because they were now installing all the libraries to `lib64` instead of `lib`.

Any ideas why this might be happening and how to solve systemically rather than overriding the `install` phase and symlinking them?

FYI Cmake that got concretized is 3.23.1 and I'm not sure what I was using before.

Cheers

**Comment by msimberg** [comment]
Do any of your packages use `GNUInstallDirs` (https://cmake.org/cmake/help/latest/module/GNUInstallDirs.html?highlight=gnuinstalldirs)? If yes, that's most likely what is installing things to `lib64` and you should most likely let that be. `GNUInstallDirs` is meant to pick the most "appropriate" directory for the given system.

Do you have a good reason to depend on things being in exactly `lib`? If you e.g. need only `find_package(...)` then CMake should already look in `lib64` depending on the system.

This is most likely not a spack issue.

**Reply by salotz** [reply]
I know its not a Spack issue per se which is why I ask here and not in an issue.

I will check for those things in the build scripts and see what is going on. I suspect that the CMake scripts for the packages this is failing on are doing some hardcoding that doesn't make it "just work".

Thanks for the response! This helps a lot.
